posthog-core/src/index.ts

```diff
@@ -855,8 +860,12 @@ export abstract class PostHogCoreStateless {
           }
         : {
             method: 'POST',
-            headers: { ...this.getCustomHeaders(), 'Content-Type': 'application/json' },
-            body: payload,
+            headers: {
+              ...this.getCustomHeaders(),
+              'Content-Type': 'application/json',
+              ...(useGzip ? { 'Content-Encoding': 'gzip' } : {}),
+            },
+            body: useGzip ? await gzipCompress(payload) : payload,
```

Comment:
we should only set `'Content-Encoding': 'gzip'` if `gzipCompress(payload)` also returns valid data.
so ideally you compute the `body` first and depending on the result, you set the encoding.